### PR TITLE
Correctly combine type tags in setting decorator to handle comments.

### LINKED ISSUE #256: setting decorator does not handle trailing comments on accepted types
The `@setting` decorator currently fails if given argument types with trailing comments. It seems to be concatenating the user-provided strings, so later types become part of the comment instead of part of the type. For example:

```python
@setting('oops', 25, a='s: the first arg', b='i')
def oops(self, c, a, b):
    pass
```

will register itself with an accepted type of `s: the first argi` rather than `si`.

### PR DESCRIPTION
**Pull request by maffoo**
Fixes #256

### CODE REVIEW

**Comment by ejeffrey** [comment]
I would rename "make_type_string" to something like "combine_type_strings".  Otherwise LGTM.

File: labrad/test/test_decorators.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class DummyServer(LabradServer):
+    @setting(100, param0='s: this is a string',
+                  param1='b: this is a boolean')
+    def dummy_setting(self, c, param0, param1):
+        pass
+
+
+def test_setting_types_with_comments():
+    accepted_tags = DummyServer.dummy_setting.accepts
+    print accepted_tags
```
Comment (DanielSank) [inline labrad/test/test_decorators.py]: In my experience, printing doesn't work well in pytest. Do you know something I don't? :)

**Comment by DanielSank** [comment]
> I would rename "make_type_string" to something like "combine_type_strings".

`combine_type_tags` seems to get to the point.

File: labrad/test/test_decorators.py (hunk 1 above)
Comment (maffoo) [inline labrad/test/test_decorators.py]: LOL, will remove, that was just for my own info while testing the test. FYI, you can see stdout by including the [`-s` option](https://pytest.org/latest/capture.html) when running pytest (not that we should get in this habit, but it's useful on occasion when debugging a test).

Comment (DanielSank) [inline labrad/test/test_decorators.py]: Ok. Beyond that LGTM.

**Pull request merged**